### ISSUE TITLE
Unable to Add SparkleXRM package, causes Erro VS2012 crm 2013 SDK .Net 4.5.2

### ISSUE THREAD

**Issue by mrmwink** [scottdurow/SparkleXrm#92]
Hi,

Great product, everything I see of it looks amazing I just want to try using it but can't get the samples to compile!

I have CRM  2013 on prem, using vs2012 with dev toolkit installed all running .net 4.5.2 and standard templates working fine. When I open the SparkleXrmTemplate it opens fine but there are loads of unresolved references in the plugins / workflows projects, basically all the crm sdk references! I can re-add these fine from my sdk location, but the one that seems to causes problems is the sparklexrm package. When I install through nuget it installs fine but instantly I get errors showing as below..

type of namespace comvisibleattribute could not be found
ComVisible, GuidAttribute, Guid, essentially the assemblyinfo file is red!

Also get the same for collection, tuple and Iserviceprovider.

Help

Mike

**Comment by scottdurow** [comment]
Hi Mike,
Have you performed a NuGet restore once you open the template project? This will download all the dependencies. You will also need to install the Script# visual studio add-in via the Visual Studio Gallery.

**Comment by scottdurow** [comment]
You often then need to build each project independently before doing a full solution build. Have you followed http://www.sparklexrm.com/s/Tutorials/SetUpNewProject.html

**Comment by mrmwink** [comment]
Yes, did nuget restore and it ignores any CRM sdk references and sparklexrm completely, it doesn't even attempt to restore it!

It does install various other projects, and weirdly if I exclude sparklexrm reference, change c# class1 file and rebuild it run the script# and generates the jscript files in the crmpackage.

I'm very confused

**Comment by mrmwink** [comment]
And yes, I have followed the video, tried building everything separately, leaving plugins until last.

I just can't understand why SparkleXrm isn't being installed as part of the restore, and why when I do so manually it breaks the plugins project only with all the reference errors, it feels like a versioning issue, like when you have different projects with different .net versions but I've checked everything there.

Just checking the CRM 2013 SDK file version for Microsoft.xrm.sdk.dll its 6.1.0.542, this is compatible I hope?

Just looking at the hint path for the dll's in the plugins project they are shown as 

<HintPath>..\..\..\..\SparkelXrm\DebugWeb\WebResources\sparkle_\References\SparkleXrm.dll</HintPath>

 <HintPath>E:\Program Files (x86)\Microsoft Visual Studio 11.0\Common7\IDE\Extensions\Microsoft\Dynamics CRM 2011 Developer Tools\1.0\Microsoft.Xrm.Sdk.dll</HintPath>

<HintPath>E:\Program Files (x86)\Microsoft Visual Studio 11.0\Common7\IDE\Extensions\Microsoft\Dynamics CRM 2011 Developer Tools\1.0\Microsoft.Crm.Sdk.Proxy.dll</HintPath>

I am using the SparkleXrm-v-.2.8/SparkleXrmSamples/SparkleXrmTemplate, this is the correct 2012 / crm 2013 solution correct?

Thanks

Mike

**Comment by mrmwink** [comment]
![capture](https://cloud.githubusercontent.com/assets/18310333/14345155/698ec754-fca3-11e5-9625-794334cde92a.PNG)

**Comment by mrmwink** [comment]
So, by not installing SparkleXRM, and instead just grabbing the DLL from an install on another project everything builds and deploys fine! In order to follow the videos I've also copied over the web resources and classes from the 2015 template as they're not in the 2011/2013 template!

**Comment by Stanzarectah** [comment]
Yes, you are correct none of the Microsoft or SparkleXrm libraries appear under packages folder for the 2013 template (no idea why) so I would just copy them from other projects

**Comment by synesthesia** [comment]
I had this same problem copying the 2015withunittests template and using on VS2012

I checked packages and all were restored

Removing following lines from ClientUI.UniTests.csproj seems to resolve:


```
  <Import Project="$(SolutionDir)\.nuget\NuGet.targets" Condition="Exists('$(SolutionDir)\.nuget\NuGet.targets')" />
  <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
    <PropertyGroup>
      <ErrorText>This project references NuGet package(s) that are missing on this computer. Enable NuGet Package Restore to download them.  For more information, see http://go.microsoft.com/fwlink/?LinkID=322105. The missing file is {0}.</ErrorText>
    </PropertyGroup>
    <Error Condition="!Exists('$(SolutionDir)\.nuget\NuGet.targets')" Text="$([System.String]::Format('$(ErrorText)', '$(SolutionDir)\.nuget\NuGet.targets'))" />
  </Target>
```
